internal/keyvault/keyvault.go

```diff
@@ -0,0 +1,91 @@
+package keyvault
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"github.com/hashicorp/vault-client-go"
+	"github.com/hashicorp/vault-client-go/schema"
+	"github.com/notaryproject/notation-hashicorp-vault/internal/crypto"
+	"os"
+	"strings"
+	"time"
+)
+
+type VaultClientWrapper struct {
+	vaultClient *vault.Client
+
+	keyID string
+}
+
+func NewVaultClientFromKeyID(id string) (*VaultClientWrapper, error) {
+	// read addr and token from environment variables
+	vaultAddr := os.Getenv("VAULT_ADDR")
+	if len(vaultAddr) < 1 {
+		return nil, errors.New("failed to load vault address")
+	}
+
+	vaultToken := os.Getenv("VAULT_TOKEN")
+	if len(vaultToken) < 1 {
+		return nil, errors.New("Error loading vault token")
+	}
+
+	// prepare a client with the given base address
+	client, err := vault.New(
+		vault.WithAddress(vaultAddr),
+		vault.WithRequestTimeout(30*time.Second),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// authenticate with a root token (insecure)
+	if err := client.SetToken(vaultToken); err != nil {
+		return nil, err
+	}
+
+	return &VaultClientWrapper{
+		vaultClient: client,
+		keyID:       id,
+	}, nil
+}
+
+func (vw *VaultClientWrapper) GetCertificateChain(ctx context.Context) ([]*x509.Certificate, error) {
+	// read a certChain
+	secret, err := vw.vaultClient.Secrets.KvV2Read(ctx, vw.keyID)
+	if err != nil {
+		return nil, err
+	}
+	certString, ok := secret.Data.Data["certificate"].(string)
+	if !ok {
+		return nil, errors.New("failed to parse certificate from KV secrets engine")
+	}
+	certBytes := []byte(certString)
+	return crypto.ParseCertificates(certBytes)
+}
+
+func (vw *VaultClientWrapper) SignWithTransit(ctx context.Context, encodedData string, signAlgorithm string) ([]byte, error) {
+	// sign with transit SE
+	resp, err := vw.vaultClient.Secrets.TransitSign(ctx, vw.keyID, schema.TransitSignRequest{
```

Comment:
One additional commentary here that I think is worth discussing: this approach hides the mount path problem currently.

See: `vault.WithMountPath("my/approle/path")` in the README: https://github.com/hashicorp/vault-client-go 

Currently this requires `kv` and `transit` plugins to be mounted at the default paths. We probably want the user to be able to specify a custom [path](https://developer.hashicorp.com/vault/docs/plugins) (including [namespaces](https://developer.hashicorp.com/vault/docs/enterprise/namespaces))... Maybe via environment variable or config file?

Reply:
Oh that makes sense. I think using a config file is a good idea.